### PR TITLE
fix: SparseArray canonicalize needs to correctly handle validity

### PR DESCRIPTION
**Pull request by lwwmanning**
... and as a bonus, it should now be much faster for sparse bools

### CODE REVIEW

File: vortex-array/src/array/sparse/canonical.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+fn canonicalize_sparse_bools(
+    values_array: BoolArray,
+    indices: &[usize],
+    len: usize,
+    fill_value: &ScalarValue,
+) -> VortexResult<Canonical> {
+    let fill_bool: bool = if fill_value.is_null() {
+        bool::default()
```
Comment (AdamGS) [inline vortex-array/src/array/sparse/canonical.rs]: nit - a `false` here is much easier to read IMO, also doesn't need the type hint in line 45

**Review by AdamGS — APPROVED** [review]
LGTM

**Review by danking — APPROVED** [review]
:shipit:

File: vortex-array/src/array/sparse/canonical.rs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    use vortex_error::VortexExpect as _;
+    use vortex_scalar::ScalarValue;
+
+    use crate::array::sparse::SparseArray;
+    use crate::array::{BoolArray, PrimitiveArray};
+    use crate::validity::Validity;
+    use crate::{ArrayDType, IntoArray, IntoCanonical};
+
+    #[rstest]
+    #[case(Some(true))]
```
Comment (robert3005) [inline vortex-array/src/array/sparse/canonical.rs:117]: In this case the validity should be AllValid and not an Array

Comment (robert3005) [reply vortex-array/src/array/sparse/canonical.rs:117]: hm, no, there's on null value

**Pull request merged**